Adopt smart pointers in WebCore/bridge/ files

mikhailramalho opened the pull request:
#### 0a2287a135f68bd03cbcee31be0afd575c27e207
<pre>
Adopt smart pointers in WebCore/bridge/ files
<a href="https://bugs.webkit.org/show_bug.cgi?id=273642">https://bugs.webkit.org/show_bug.cgi?id=273642</a>

Reviewed by Chris Dumez.

Changes based on clang-tidy&apos;s alpha.webkit.UncountedLocalVars checker.

* Source/WebCore/bridge/runtime_array.cpp:
(JSC::JSC_DEFINE_CUSTOM_GETTER):
(JSC::RuntimeArray::getOwnPropertyNames):
(JSC::RuntimeArray::getOwnPropertySlot):
(JSC::RuntimeArray::put):
(JSC::RuntimeArray::putByIndex):
* Source/WebCore/bridge/runtime_method.cpp:
(JSC::JSC_DEFINE_CUSTOM_GETTER):
(JSC::RuntimeMethod::getOwnPropertySlot):
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/WebCore/bridge/runtime_object.cpp:
(JSC::Bindings::JSC_DEFINE_CUSTOM_GETTER):
(JSC::Bindings::RuntimeObject::getOwnPropertySlot):
(JSC::Bindings::RuntimeObject::put):
(JSC::Bindings::JSC_DEFINE_HOST_FUNCTION):
(JSC::Bindings::RuntimeObject::getOwnPropertyNames):

Canonical link: <a href="https://commits.webkit.org/278340@main">https://commits.webkit.org/278340@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/814716daac90eb22b33582922a6eb3672d219bd1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50244 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29537 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2541 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53503 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/934 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52547 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35764 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/565 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/40994 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52343 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27205 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43245 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22093 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/24623 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/497 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8622 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/46609 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/559 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55088 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25340 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/495 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/48396 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/26601 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/43430 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47419 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11020 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27465 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/26333 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/bridge/runtime_object.cpp:
```diff
@@ -182,23 +179,22 @@ bool RuntimeObject::getOwnPropertySlot(JSObject* object, JSGlobalObject* lexical
         }
     }
-        
+
```
Please update your IDE to stop doing this. This is frowned upon as it adds unnecessary noise to patches.

mikhailramalho replied:
I actually did it manually... I thought it wouldn't hurt to get rid of these whitespace since this was a small patch. I'll update the PR.

mikhailramalho committed:
Adopt smart pointers in WebCore/bridge/ files
https://bugs.webkit.org/show_bug.cgi?id=273642

Reviewed by Chris Dumez.

Changes based on clang-tidy's alpha.webkit.UncountedLocalVars checker.

* Source/WebCore/bridge/runtime_array.cpp:
(JSC::JSC_DEFINE_CUSTOM_GETTER):
(JSC::RuntimeArray::getOwnPropertyNames):
(JSC::RuntimeArray::getOwnPropertySlot):
(JSC::RuntimeArray::put):
(JSC::RuntimeArray::putByIndex):
* Source/WebCore/bridge/runtime_method.cpp:
(JSC::JSC_DEFINE_CUSTOM_GETTER):
(JSC::RuntimeMethod::getOwnPropertySlot):
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/WebCore/bridge/runtime_object.cpp:
(JSC::Bindings::JSC_DEFINE_CUSTOM_GETTER):
(JSC::Bindings::RuntimeObject::getOwnPropertySlot):
(JSC::Bindings::RuntimeObject::put):
(JSC::Bindings::JSC_DEFINE_HOST_FUNCTION):
(JSC::Bindings::RuntimeObject::getOwnPropertyNames):

Canonical link: https://commits.webkit.org/278340@main

wkia commented on Source/WebCore/bridge/runtime_array.cpp:
```diff
@@ -122,11 +121,11 @@ bool RuntimeArray::getOwnPropertySlotByIndex(JSObject* object, JSGlobalObject* l
 bool RuntimeArray::put(JSCell* cell, JSGlobalObject* lexicalGlobalObject, PropertyName propertyName, JSValue value, PutPropertySlot& slot)
 {
-    VM& vm = lexicalGlobalObject->vm();
+    Ref vm = lexicalGlobalObject->vm();
```
Didn't it break debug build with `C_LOOP` enabled? 

The following macro should fail in Win32 Debug:
```
#define EXCEPTION_SCOPE_POSITION_FOR_ASAN(vm__) (vm__).currentCLoopStackPointer()
```
something like 
```
error: no member named 'currentCLoopStackPointer' in 'WTF::Ref<JSC::VM>'; did you mean to use '->' instead of '.'?
```
